edge-23.3.2 (#10523)
linkerd/linkerd2: CHANGES.md

# Changes

## edge-23.3.2

This edge release continues to improve dynamic Policy statuses and
introduces support for header-based routing.

* Destination Controller

>   * Added OutboundPolices API, for use by `linkerd-proxy` to route

Reviewer: ```suggestion
  * Added OutboundPolicies API, for use by `linkerd-proxy` to route
```

    outbound traffic
  * Improved diagnostic log messages
  * Fixed sending of spurious profile updates

* Proxy

>   * Utilize new OutboundPolices API, supporting Gateway API-style routes

Reviewer: ```suggestion
  * Use the new OutboundPolicies API, supporting Gateway API-style routes
```

# Changes

## edge-23.3.2

This edge release continues to improve dynamic Policy statuses and
introduces support for header-based routing.

* Destination Controller
  * Added OutboundPolices API, for use by `linkerd-proxy` to route
    outbound traffic
  * Improved diagnostic log messages
  * Fixed sending of spurious profile updates

* Proxy
  * Utilize new OutboundPolices API, supporting Gateway API-style routes
    in the outbound proxy

>   * Proxies in ingress and gateway configurations may now

Reviewer 2: This may be too much in the weeds for release notes.  Does it affect users?

Author: Not certain if you meant lines 15/16 or 17.

Author: I see, you meant 17. Removing it.

>   * Proxies in ingress and gateway configurations may now

Reviewer: * Use the new OutboundPolicies API to discover Gateway API-style
    HTTPRoute configurations

>   * Utilize new OutboundPolices API, supporting Gateway API-style routes
>     in the outbound proxy
>   * Proxies in ingress and gateway configurations may now
>     discover policies by name instead of network address

Reviewer: ```suggestion
  * Use the new OutboundPolicies API to discover Gateway API-style
    HTTPRoute configurations to support arbitrary request
    routing.
```

* Policy Controller

>   * Utilize `policy-controller-write` Lease to avoid multiple writers

Reviewer 2: Do we want to phrase this in terms of being able to run multiple replicas of the policy controller / run it in HA mode?

Author: Good idea, I'm reframing

    when patching HTTPRoutes
  * Consider the `status` field and filter out HTTPRoutes which have not
    been accepted

>   * Controller name value has been renamed from

Reviewer 2: Where is this used? In the lease resource?  This note makes it sound like the deployment name has changed.

Author: This line isn't even necessary because this is the first release using a Lease. I'm going to remove it.

    `policy.linkerd.io/status-controller` to `linkerd.io/policy-controller`

> * Added KubeAPI server ports to `ignoreOutboundPorts` of `proxy-injector`

Reviewer 2: Did this fix an issue, or was it just an oddity?

Author: From the PR introducing this:

> To avoid startup issues in the proxy-injector, the same configuration
    should be applied.

Do you think I should update the message to reflect that this addresses startup issues?
